src/npcmove.cpp

```diff
@@ -1368,7 +1368,11 @@ void npc::move()
         action = method_of_fleeing();
     } else if( ( target == &player_character && attitude == NPCATT_FLEE_TEMP ) ||
                has_effect( effect_npc_run_away ) ) {
-        action = method_of_fleeing();
+        if( rl_dist( pos(), player_character.pos() ) <= 1 ) {
```

Comment:
```suggestion
        if( target && rl_dist( pos(), target.pos() ) <= 1 ) {
```

They could have `effect_npc_run_away` but not be targeting the player.